src/apps/Settings/HomeSettings.js

```diff
@@ -0,0 +1,169 @@
+
+class HomeSettings extends React.Component {
+  static propTypes = {
+    account: EthereumAddressType,
+    apps: PropTypes.arrayOf(AppType).isRequired,
+    walletWeb3: PropTypes.object.isRequired,
+  }
+  state = {
+    homeAppName: 'Home',
+    selectedHomeAppName: defaultName,
+    storageApp: null,
+  }
+
+  async getHomeSettings(props) {
+    const { apps, walletWeb3, account } = props
+    const storageApp = apps.find(({ name }) => name === 'Storage')
+    let homeAppName, homeAppAddr
+
+    // TODO: Move this code to a subscriber
+    if (storageApp && storageApp.proxyAddress) {
+      homeAppAddr = await AragonStorage.get(
+        walletWeb3,
+        storageApp.proxyAddress,
+        account,
+        'HOME_APP'
+      )
+
+      homeAppName = await AragonStorage.get(
+        walletWeb3,
+        storageApp.proxyAddress,
+        account,
+        'HOME_APP_NAME'
+      )
+      if (!homeAppName) {
+        homeAppName = 'Home'
+      }
+    }
+    const selectedHomeApp = apps.find(
+      ({ proxyAddress }) => proxyAddress === homeAppAddr
+    )
+
+    this.setState({
+      homeAppName: homeAppName,
+      selectedHomeAppName:
+        (selectedHomeApp && selectedHomeApp.name) || defaultName,
+      storageApp: apps.find(({ name }) => name === 'Storage'),
+    })
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.getHomeSettings(nextProps)
+  }
+  componentWillMount() {
+    this.getHomeSettings(this.props)
+  }
+
+  handleHomeAppChange = (index, apps) => {
+    this.setState({ selectedHomeAppName: apps[index] })
+  }
+
+  handleHomeNameChange = event => {
+    this.setState({
+      homeAppName: event.target.value && event.target.value.trim(),
+    })
+  }
+
+  handleHomeSettingsSave = async () => {
+    const { walletWeb3, apps, account } = this.props
+    const { storageApp, selectedHomeAppName, homeAppName } = this.state
+
+    if (storageApp && storageApp.proxyAddress) {
+      const selectedAppAddr = apps.find(
+        ({ name }) => name === selectedHomeAppName
+      )
+
+      try {
+        if (selectedHomeAppName !== defaultName) {
+          await AragonStorage.set(
+            walletWeb3,
+            storageApp.proxyAddress,
+            account,
+            'HOME_APP_NAME',
+            homeAppName
+          )
+        }
+
+        await AragonStorage.set(
+          walletWeb3,
+          storageApp.proxyAddress,
+          account,
+          'HOME_APP',
+          (selectedAppAddr && selectedAppAddr.proxyAddress) || ''
+        )
+        window.location.reload(true)
+      } catch (err) {
+        console.log(err)
+      }
+    }
+  }
+
+  render() {
+    const { apps } = this.props
+    const { homeAppName, selectedHomeAppName, storageApp } = this.state
+
+    const reducedAppNames = apps.reduce(
+      (filtered, app) => {
+        app.hasWebApp && filtered.push(app.name)
+        return filtered
+      },
+      [defaultName]
+    )
+    if (!storageApp) return <div />
```

Comment:
instead of the negative condition, it seems more clear to use a positive condition, or a ternary like this:
`return ({storageApp ? <Option>etc</Option> : null})`

Reply:
I tried that approach before ending up doing this way to solve one stylesheet bug.
```return (<div>{storageApp ? <Option>etc</Option> : null}</div>)```
With this approach, the style of the app breaks. It needs to have a <Options> component like the very first component. And if you set it like this:
```return (<Option>{storageApp ? etc : null}</Option>)```
You end up with an unwanted margins when HomeSettings are not showing up.